### PR TITLE
Add files to a Work's change history

### PR DESCRIPTION
**Pull request by rschenk**
One thing I thought about doing but decided against, was making a base class for both WorkVersionChangePresenter and FileMembershipChangePresenter to inherit from. They do share a decent amount of code, but... `¯\_(ツ)_/¯`

### CODE REVIEW

File: app/presenters/file_membership_change_presenter.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+
+    case event
+    when 'create' then t('create')
+    when 'destroy' then t('destroy')
+    else "#{event}d".titleize
+    end
+  end
+
+  def timestamp
+    created_at.to_s(:long)
```
[the remark this answers is not in the document]
Comment (rschenk) [reply app/presenters/file_membership_change_presenter.rb]: Yup, I think this is a good idea, and also the one thing I really wanted to get out of making these Presenters both inherit from a super class. 

You can define your own system-wide date formats with Rails:
https://api.rubyonrails.org/classes/DateTime.html#method-i-to_formatted_s

File: app/presenters/file_membership_change_presenter.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    'file_membership_change'
+  end
+
+  def action
+    return t('rename') if rename?
+
+    case event
+    when 'create' then t('create')
+    when 'destroy' then t('destroy')
+    else "#{event}d".titleize
```
[the remark this answers is not in the document]
Comment (rschenk) [reply app/presenters/file_membership_change_presenter.rb]: Yeah good thinking. I think I'll go ahead and do this. And then if we do also add a thing for `'update'` then there's no need for this case statement

File: app/presenters/work_history_presenter.rb [hunk 3]
```diff
@@ -43,14 +43,30 @@ def load_changes
...
+                user: lookup_user(paper_trail_version.whodunnit)
+              )
+            end
+
+          all_changes = (
+            changes_to_work_version +
+            changes_to_files
+          ).sort_by(&:created_at)
+
+          [decorated_work_version, all_changes]
```
[the remark this answers is not in the document]
Comment (rschenk) [reply app/presenters/work_history_presenter.rb]: maybe what I'll do is split `changes_from_work_version` and `changes_from_files` into their own methods. But I also wanted to keep them together so it was obvious that they were both returning Presenters...

File: app/models/file_version_membership.rb [hunk 4]
```diff
@@ -16,6 +16,17 @@ class FileVersionMembership < ApplicationRecord
 
   delegate :size, :mime_type, :original_filename, to: :uploader
 
+  attr_accessor :changed_by_system
```
[the remark this answers is not in the document]
Comment (rschenk) [reply app/models/file_version_membership.rb]: `nil` is falsey, yes, which is why I didn't initialize it

File: spec/presenters/work_history_presenter_spec.rb [hunk 5]
```diff
@@ -36,9 +36,10 @@
     end
 
     it 'returns a 2d array, where dimension 2 is the changes to each version, mapped to Presenter objects' do
-      _version, changes = changes_by_work_version.first
-      expect(changes).not_to be_empty
-      expect(changes).to all(be_a(WorkVersionChangePresenter))
+      _published_version, changes = changes_by_work_version.find { |v, _| v == v1 }
```
[the remark this answers is not in the document]
Comment (rschenk) [reply spec/presenters/work_history_presenter_spec.rb]: What I'm testing in this specific test is that the array of changes include both `WorkVersionChangePresenters` _and_ `FileMembershipChangePresenters`. And only _published_ works are guaranteed to have Files, and therefore also `FileMembershipChangePresenters`. The draft version won't necessarily have any files, so it's not useful for this test.

File: app/presenters/file_membership_change_presenter.rb (hunk 1 above)
[the remark this answers is not in the document]
Comment (awead) [reply app/presenters/file_membership_change_presenter.rb]: @rschenk we can ticket the DateTime thing. As far as inheriting (I see your comment on the actual PR), we can hold off for the time being, maybe ticket it as well, and revisit later.

**Pull request merged**